queue_processors/autoreduction_processor/post_process_admin_utilities.py

```diff
@@ -0,0 +1,82 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Post-Process Admin Utilities
+"""
+
+import sys
+from contextlib import contextmanager
+
+
+class SkippedRunException(Exception):
+    """
+    Exception for runs that have been skipped
+    Note: this is currently only the case for EnginX Event mode runs at ISIS
+    """
+
+
+@contextmanager
+def channels_redirected(out_file, error_file, out_stream):
+    """
+    This context manager copies the file descriptor(fd) of stdout and stderr to the files given in
+    out_file and err_file respectively. The fd is at the C level and so picks up data sent via
+    Mantid. Both output streams are additionally also sent to out_stream.
+    :param out_file: (str) output path
+    :param error_file: (str) error file path
+    :param out_stream: (class object) io.StringIO
+    """
+
+    old_stdout, old_stderr = sys.stdout, sys.stderr
+
+    class MultipleChannels:
+        # pylint: disable=expression-not-assigned
+        """ Behaves like a stream object, but outputs to multiple streams."""
+
+        def __init__(self, *streams):
+            """
+            :param streams: (tuple) (N-path, io.StringIO)
+            """
+            self.streams = streams
+
+        def write(self, stream_message):
+            """ Write to steams.
+            :param stream_message: (data) stream message
+            """
+            [stream.write(stream_message) for stream in self.streams]
+
+        def flush(self):
+            """ Flush streams. """
+            [stream.flush() for stream in self.streams]
+
+    def _redirect_channels(output_file, err_file):
```

Comment:
Question mark

Reply:
Thanks, taken out this time. 🙂  Left in origionally as I beleive this was an unsurety about whether or not this is actually the redirected channels or not